entities/IoCContainer.java

```diff
@@ -0,0 +1,51 @@
+package entities;
+
+import exceptions.CustomExceptions;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/*
+Lines for use the container
+1. IoCContainer ioc = IoCContainer.getInstance();
+2. ioc.Register(Entitie_name,new Entitie_name());
+3. Entitie_name any_name = (Entitie_name)ioc.Resolve("Entitie_name");
+4. Now you can use the name used in any_name for to the methodes of the entitie
+ */
+public class IoCContainer<T> {
+
+    //Singleton to instantiate the IoCContainer only once
+    private static IoCContainer ioc;
+    private IoCContainer(){}
+    public static IoCContainer getInstance(){
+        if(ioc == null){
+            ioc = new IoCContainer();
+        }return ioc;
```

Comment:
I think you can use optionals here as well:
```
return Optional.ofNullable(ioc).orElse(IoCContainer::new)
```

Reply:
done, with a little adjust orElseGet